persistence/src/main/java/edu/unc/lib/dl/persist/services/destroy/DestroyObjectsService.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2008 The University of North Carolina at Chapel Hill
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.persist.services.destroy;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import edu.unc.lib.dl.acl.service.AccessControlService;
+import edu.unc.lib.dl.acl.util.AgentPrincipals;
+import edu.unc.lib.dl.acl.util.Permission;
+import edu.unc.lib.dl.fcrepo4.PIDs;
+import edu.unc.lib.dl.fedora.PID;
+
+/**
+ * Service that manages the destruction of objects in the repository and their replacement by tombstones
+ *
+ * @author harring
+ *
+ */
+public class DestroyObjectsService {
+
+    private static final Logger log = LoggerFactory.getLogger(DestroyObjectsService.class);
+
+    private AccessControlService aclService;
+
+    /**
+     * Checks whether the active user has permissions to destroy the listed objects,
+     * and then kicks off a job to destroy the permitted ones asynchronously
+     *
+     * @param agent security principals of the agent making request
+     * @param ids list of objects to destroy
+     */
+    public void destroyObjects(AgentPrincipals agent, List<String> ids) {
+        List<PID> objsToDestroy = new ArrayList<>();
+
+        for (String id : ids) {
+            PID pid = PIDs.get(id);
+            aclService.assertHasAccess("User does not have permission to destroy this object", pid,
```

Comment:
Annoyingly, destroy and destroyUnit are two separate permissions right now since they are assigned to different roles. So you will need to figure out if the object is a unit to know which permission to check for.

Reply:
ok, good to know. how would you like us to handle any AccessRestrictionExceptions that may be thrown? is it enough to eat them, or would you like to log them? obviously we don't want to blow up the for-loop, so i went ahead and wrapped those checks in a try-catch

Reply:
actually i'm not realizing this loop is just over the top level objects being deleted, not for every child object that will be deleted. 

The access restriction exceptions need to be reported in some way, preferably to the client that made the request. I don't think we need to log them. My first inclination is to block the whole destroy operation if the user doesn't have permission to destroy any of the objects being requested (since it was submitted as a single operation), but I'm open to continuing with the other objects if you think that makes more sense. I'm not sure how we are handling this situation in other operations right now.

Reply:
I think I'll just leave the calls to aclService.assertHasAccess() as they are, and remove the try-catch stuff. That way the action will get canceled right away and whatever client calls destroyObjects() on the service can handle the AccessRestrictionException. Sound reasonable?